iothub_client/inc/iothub_client_properties.h

```diff
@@ -0,0 +1,300 @@
+
+typedef struct IOTHUB_CLIENT_REPORTED_PROPERTY_TAG {
+    /** @brief   Version of the structure.  Currently must be IOTHUB_CLIENT_REPORTED_PROPERTY_VERSION_1. */
+    int structVersion;
+    /** @brief    Name of the property. */
+    const char* name;
+    /** @brief    Value of the property. */
+    const char* value;
+} IOTHUB_CLIENT_REPORTED_PROPERTY;
+
+/** @brief Current version of @p IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE structure.  */
+#define IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE_VERSION_1 1
+
+/** @brief    This struct represents the response to a writable property that the device will return.  
+*             This structure is filled out by the application and can be serialized into a payload for IoT Hub via
+*             @p IoTHubClient_Serialize_WritablePropertyResponse(). */
+typedef struct IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE_TAG {
+    /** @brief   Version of the structure.  Currently must be IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE_VERSION_1. */
+    int structVersion;
+    /** @brief Name of the property. */
+    const char* name;
+    /** @brief Value of the property. */
+    const char* value;
+    /** @brief Result of the requested operation.  This maps to an HTTP status code.  */
+    int result;
+    /** @brief Acknowledgement version.  This corresponds to the version of the writable properties being responded to. */
+    /** @details If using @p IoTHubClient_Deserialize_Properties_CreateIterator() to deserialize initial property request
+    *            from IoT Hub, just set this field to what was returned in @p propertiesVersion. */
+    int ackVersion;
+    /** @brief Optional friendly description of the operation.  May be NULL. */
+    const char* description;
+} IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE;
+
+/** @brief Enumeration that indicates whether a given property from the service was originally reported from the device
+           (and hence the device application sees what the last reported value IoT Hub has)
+           or whether this is a writable property that the service is requesting configuration for.
+*/
+typedef enum IOTHUB_PROPERTY_TYPE_TAG 
+{
+    /** @brief Property was initially reported from the device. */
+    IOTHUB_CLIENT_PROPERTY_TYPE_REPORTED_FROM_DEVICE,
+    /** @brief Property is writable.  It is configured by service remotely. */
+    IOTHUB_CLIENT_PROPERTY_TYPE_WRITABLE
+} IOTHUB_CLIENT_PROPERTY_TYPE;
+
+/* @brief Enumeration that indicates whether the JSON value of a deserialized property 
+*         is a null-terminated string or binary.
+*/
+typedef enum IOTHUB_CLIENT_PROPERTY_VALUE_TYPE_TAG
+{
+    /* @brief Deserialized JSON value is a string. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_STRING,
+    /* @brief Deserialized JSON value is a binary. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_BINARY
+} IOTHUB_CLIENT_PROPERTY_VALUE_TYPE;
+
+/** @brief Current version of @p IOTHUB_CLIENT_DESERIALIZED_PROPERTY structure.  */
+#define IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1 1
```

Comment:
Would it make sense to rename to `IOTHUB_CLIENT_DESERIALIZED_PROPERTY_STRUCT_VERSION_1` ?